File: src/Cpp/09_Listy_wprowadzenie/Zad17.cpp
#include <algorithm>
#include <cassert>
#include <vector>

//Dla otrzymanej listy oraz liczby X, znajdz indeksy pierwszej
//pary elementów listy, ktorych suma jest rowna liczbie X.
//Jesli w liscie nie ma takiej pary, zwroc pare (-1, -1)

std::pair<int, int> sumaDwochV1(std::vector<int>& lista, int liczba)
{
    int n = lista.size();

    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            if (lista[j] + lista[i] == liczba)
                return std::pair<int, int>(i, j);
        }
    }

    return std::pair<int, int>(-1, -1);
}

void test1()
{
    std::vector<int> lista{ 1, 3, 4, 5, 2 };
    int liczba = 5;
    std::pair<int, int> wynik(0, 2);
    assert(sumaDwochV1(lista, liczba) == wynik);
}

void test2()
{
    std::vector<int> lista{ 2, -6, 10, 21, 8 };
    int liczba = 7;
    std::pair<int, int> wynik(-1, -1);
    assert(sumaDwochV1(lista, liczba) == wynik);
}

int main()
{

    test1();
    test2();

    return 0;
}

File: src/JavaScript/09_Listy_wprowadzenie/Zad17.js
var Pair = /** @class */ (function () {
    function Pair(a, b) {
        this.first = a;
        this.second = b;
    }
    return Pair;
}());

var Main = (function () {
    function Main() {
    }
    Main.sumaDwochV1 = function (lista, liczba) {
        var n = lista.length;
        for (var i = 0; i < n; i++) {
            {
                for (var j = i + 1; j < n; j++) {
                    {
                        if ( lista[j] +  lista[i] === liczba) {
                            return (new Pair(i, j));
                        }
                    };
                }
            };
        }
        return (new Pair(-1, -1));
    };
    Main.test1 = function () {
        var lista = [1, 3, 4, 5, 2]
        var liczba = 5;
        var oczekiwane = new Pair(0, 2)
       var wynik = Main.sumaDwochV1(lista, liczba);
        if (!(wynik.first == oczekiwane.first && wynik.second == oczekiwane.second)) {
            throw new Error(`Assertion error line 34: ${wynik.first} != ${oczekiwane.first} || ${wynik.second} != ${oczekiwane.second}`);
        }
    };
    Main.test2 = function () {
        var lista = [2, -6, 10, 21, 8]
        var liczba = 7;
        var oczekiwane = new Pair(-1, -1)
        var wynik = Main.sumaDwochV1(lista, liczba);
        if (!(wynik.first == oczekiwane.first && wynik.second == oczekiwane.second)) {
            throw new Error(`Assertion error line 34: ${wynik.first} != ${oczekiwane.first} || ${wynik.second} != ${oczekiwane.second}`);
        }
    };
    Main.main = function (args) {
        Main.test1();
        Main.test2();
    };
    return Main;
}());
Main["__class"] = "Main";
Main.main(null);
